launchlib/launcher.go

```diff
@@ -265,3 +282,37 @@ func createReplacer() *strings.Replacer {
 func delim(str string) string {
 	return fmt.Sprintf("%s%s%s", TemplateDelimsOpen, str, TemplateDelimsClose)
 }
+
+func filterHeapSizeArgs(args []string) []string {
+	var filtered []string
+	var hasMaxRAMPercentage, hasInitialRAMPercentage bool
+	for _, arg := range args {
+		if !isHeapSizeArg(arg) {
+			filtered = append(filtered, arg)
+		}
+
+		if isMaxRAMPercentage(arg) {
+			hasMaxRAMPercentage = true
+		} else if isInitialRAMPercentage(arg) {
+			hasInitialRAMPercentage = true
+		}
+	}
+
+	if !hasInitialRAMPercentage && !hasMaxRAMPercentage {
+		filtered = append(filtered, "-XX:InitialRAMPercentage=80.0")
+		filtered = append(filtered, "-XX:MaxRAMPercentage=80.0")
```

Comment:
@jkozlowski I hadn't actually realized this PR involves making go-java-launcher introspect the command line args defined in launcher-static.yml and replace out some of them.  I think this is honestly gonna confuse the crap out of people, because up until now the contract of go-java-launcher has been that it effectively just invokes the `java` binary with all the static & custom args laid out afterwards.

What do you think about having an extra section in launcherCustom or launcher-custom.container.yml or something that you'd expect folks to populate?

Reply:
Discussed in person, this approach is still safest and cleanest out of all the proposed solutions.